packages/platform-browser/lazy-animations/src/async_animation_renderer.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import type {ɵAnimationRendererFactory as AnimationRendererFactory, ɵcreateEngine as createEngine, ɵAnimationRenderer as AnimationRenderer} from '@angular/animations/browser';
+import {Injectable, NgZone, Renderer2, RendererFactory2, RendererStyleFlags2, RendererType2} from '@angular/core';
+
+type FromAnimationBrowserModule = {
+  ɵcreateEngine: typeof createEngine,
+  ɵAnimationRendererFactory: typeof AnimationRendererFactory
+};
+
+@Injectable()
+export class AsyncAnimationRendererFactory implements RendererFactory2 {
+  private _moduleLoaded: Promise<AnimationRendererFactory>|null = null;
+
+  constructor(
+      private doc: Document, private delegate: RendererFactory2, private zone: NgZone,
+      private animationType: 'animations'|'noop',
+      private moduleImpl?: Promise<FromAnimationBrowserModule>) {}
+
+  /**
+   * @internal
+   */
+  private loadImpl(
+      type: 'animations'|'noop',
+      moduleImpl: Promise<FromAnimationBrowserModule> = import('@angular/animations/browser')):
+      Promise<AnimationRendererFactory> {
+    return moduleImpl
```

Comment:
nit: for readability, I'd propose avoid using a dynamic import as a default value and assign the default value for the `moduleImpl` at the very beginning of the file:
```suggestion
  private loadImpl(
      type: 'animations'|'noop',
      moduleImpl?: Promise<FromAnimationBrowserModule>): Promise<AnimationRendererFactory> {
    moduleImpl ??= import('@angular/animations/browser');
    return moduleImpl
```

Reply:
We also accept the `moduleImpl` as a constructor argument. Are we supposed to use it here if it was not provided into this function as an argument? If no, we can probably rename the argument to avoid possible confusion.